content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +162,67 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
+
+For every `dvc.yaml` file, a corresponding `dvc.lock` (YAML) file is created or
+updated by certain DVC commands (`dvc run`, `dvc repro`, `dvc commit`). It
+describes the latest contents of the files involved in the pipeline(s). It has
+several purposes:
+
+- Basic tracking of intermediate and final results of a pipeline (similar to
+  [`.dvc` files](#dvc-files))
+- Allow DVC to detect when stage definitions, or their dependencies have
+  changed. Such conditions invalidate states, requiring their reproduction (see
+  `dvc status`, `dvc repro`).
+- Provides a mapping of file and directory paths tracked in the
+  <abbr>project</abbr> to their locations in the <abbr>cache</abbr> or
+  [remote storage](/doc/command-reference/remote). This is needed internally by
+  certain DVC commands to operate, such as `dvc checkout`, `dvc get`, and
+  `dvc import`.
+
+Here's an example `dvc.lock` based on the one in [`dvc.yaml`](#dvcyaml-file)
+above:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
+    deps:
+      - path: data/clean
+        md5: d8b874c5fa18c32b2d67f73606a1be60
+    params:
+      params.yaml:
+        levels.no: 5
+    outs:
+      - path: features
+        md5: 2119f7661d49546288b73b5730d76485
+      - path: performance.json
+        md5: ea46c1139d771bfeba7942d1fbb5981e
+      - path: logs.csv
+        md5: f99aac37e383b422adc76f5f1fb45004
+```
+
+Stage commands are listed again in `dvc.lock`, in order to know when their
+definitions change in the `dvc.yaml` file.
+
+Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
+(including [metrics](/doc/command-reference/metrics) and
+[plots](/doc/command-reference/plots) files) are also listed (per stage) in
+`dvc.lock`, but with an additional field to store the hash value of each file or
+directory tracked by DVC:
+
+- `md5` is the most common type of file hash, used for local file system
+  dependencies, as well as SSH <abbr>external dependencies</abbr>.
+- `etag` is used for HTTP, S3, Azure, and Google Cloud external dependencies.
+- `checksum` is only used for HDFS external dependencies.
+
+> Refer to [External Dependencies](/doc/user-guide/external-dependencies) for
+> more information.
+
+[Parameter](/doc/command-reference/params#examples) key/value pairs are listed
+separately under `params`, grouped by parameters file. (No hash value is needed
```

Comment:
`(No has  ` part can be confusing and probably is not needed

Reply:
Removed.